radiomanager/radiomanager/api/endpoints/handshaking.py

```diff
@@ -0,0 +1,72 @@
+"""
+radiomanager.api.endpoints.handshaking
+======================================
+
+Endpoints for pinging the backend, provisioning, and registering/deregistering
+"""
+
+import uuid
+import re
+from werkzeug.exceptions import BadRequest
+from flask import request
+from flask_restful import reqparse, Resource
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.serialization import load_pem_public_key
+from cryptography.exceptions import UnsupportedAlgorithm
+from ... import FlaskExtensions
+from ...models.device import Device
+
+
+class Ping(Resource):
+    """
+    An example endpoint, useful for testing purposes.
+    """
+
+    def get(self):
+        return "ping"
+
+    def post(self):
+        data = request.form["data"]
+        if data == "ping":
+            return "pong"
+        elif data == "pong":
+            return "ping"
+        else:
+            raise BadRequest("Expected \"data=ping\" or \"data=pong\"")
+
+
+class Provision(Resource):
+    """
+    Clients use this to get setup on the system
+    """
+    POST_PARSER = reqparse.RequestParser()
+    POST_PARSER.add_argument("name", type=str)
+    POST_PARSER.add_argument("public_key", type=str)
+    ALPHANUMERIC_WITH_SPACES = re.compile("([a-z]|[A-Z]|[0-9]| )*")
+
+    def post(self):
+        """
+        POSTing to this endpoint provisions a new client with no authentication
+        capabilities. The client is expected to pass a UUID.
+        """
```

Comment:
Curious on how UUID works, how will the device always generate a truly unique ID without knowing the other radios?

Reply:
The UUID bit should be removed. The server should assign the client some sort of unique identifier upon a successful provision.

Reply:
I've decided to actually implement the UUID field as well as enforce that it is unique. In practice, UUIDs are basically collision free (key space is too big), but there's a protection (with a test) included anyway.